File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    list = [];
    for(let i = 0; i < s.length; i++){
        if(s[i] == "("){
            list.push(s[i]);
        }else{
            if (list.length === 0) return false;
            list.pop();
        }
    }
    return list.length == 0;
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
class Solution {
    boolean solution(String s) {
        int parens = 0;
        for(int i = 0; i < s.length(); i++){
            if(s.charAt(i) == '(') {
                parens++;
            } else {
                parens--;
                if(parens < 0) {
                    return false;
                }
            }
        }
        return parens == 0;
    }
}